email-address-encoder.php

```diff
@@ -97,8 +97,13 @@ function eae_register_shortcode() {
 function eae_shortcode( $attributes, $content = '' ) {
     // override encoding function with the 'eae_method' filter
     $method = apply_filters( 'eae_method', 'eae_encode_str' );
-
-    return $method( $content );
+    $res = $method( $content ); // result ignoring attributes
+    $link = $attributes['link'];
```

Comment:
This will result in php notices. You have to use the `shortcode_atts()` function.

- https://developer.wordpress.org/reference/functions/shortcode_atts/
- https://developer.wordpress.org/plugins/shortcodes/shortcodes-with-parameters/

Reply:
copy that.
Have a look at the recent commit... better?

Reply:
Yeah, I cleaned it up for you. Can you test that code, see if it works for your use cases?

Reply:
I checked your cleaned up version and it works fine for me.